app/services/document_capture_session_result.rb

```diff
@@ -26,9 +26,9 @@ def selfie_status
     self[:selfie_status].to_sym
   end
 
-  def success_status
+  def success_status(selfie_required:)
     # doc_auth_success : including document, attention_with_barcode and id type verification
-    !!doc_auth_success && selfie_status != :fail && !!pii
+    !!doc_auth_success && (selfie_required ? selfie_status == :success : true) && !!pii
```

Comment:
What was the motivation for changing `selfie_status != :fail` to `(selfie_required ? selfie_status == :success : true)`?  

I’d expect `selfie_status` to equal `:not_processed` whenever `selfie_required` equals false, which makes me expect that the previous implementation would still work.

Reply:
with the presence of selfie_required argument we  now have the ability to be more strict which is a more secure to way to say what is a pass as opposed to allowing anything that is !:fail  to be a pass.  it also safeguards against the potential for values of selfie_status  to expand over time and requires us to be explicit about what passes.

Reply:
> it also safeguards against the potential for values of selfie_status to expand over time

Since we determine the values of `selfie_status` ourselves, I don’t think there’s as much of a need to safeguard against the value of `selfie_status` changing over time.  I think that would be a bigger concern if we were determining success based solely on a value that is passed to us from the vendor, like `selfie_result`.

Both implementations work.  I do think the previous implementation is more readable, but YMMV.

Reply:
it's a small add but I  do understand your readability sentiment 👍🏿 

- similar to this scenario where a selfie_status is returned when not requested, what if a selfie_status is not returned when requested
-  i think it is safer to be explicit as to what is a pass, as opposed to what is not a fail.
- you'r right, we do determine the value of selfie_status (I say that with the assumption you are speaking about the TrueIDResponse.selfie_status) ... if in a future state a new key option is added to define selfie_status, a developer must remember to  redefine what is a fail here `selfie_status != :fail` or we could run the risk of passing folks through that should not go through

something to consider 🤔  ... An easier and more readable option could be to simply use the :success attribute since it has all of these scenarios baked in and this bug nor [the previous bug](https://cm-jira.usa.gov/browse/LG-12370) would be an issue 😐

Reply:
I think may be we should create a response class for selfie required scenario(not now), where it's unambiguous and can decide `success` accordingly and from session result perspective it's simple and straight.

Reply:
In retrospective and a little bit of topic, the `session_result` should perform it's designed purpose, store business `result`, more complex logic should be in service layer(seems what the `api_image_upload_form` does). The `api_image_upload_form`  does input validation and is what a `form` should do. It also does some result validation(pii info using another form, yes it's handy to use form for validation in Rails), which sounds a little bit off.  
Since we introduce selfie related function, it feels like we are bubbling up some of the logic here, not specific in this PR, but previous ones also.

Reply:
(I’m not sure the status of this work because I was out yesterday and the jira ticket is in the review column, but I figured I should still respond.)

As discussed in past meetings, I will defer to the rest of the team about using the success attribute.  I get the argument that you’d rather be explicit about what is a pass, as opposed to what is not a fail.  I think my concern is similar to what Zach posted, that this seems like a really unexpected place to pass a param.